### DISCUSSION TITLE
Return values from Terminal

### DISCUSSION THREAD

**Discussion by dmv1** [ttscoff/bunch-docs#323] [General]
I am just getting started with Bunch and trying to automate the creation of a new virtual environment for python using Bunch. I would like to Open, Create or Remove a virtual environment. Is there a way to cd into a path (my main python project folder) and return the value of ls for the dir currently active in terminal? I would like to use the filename return to be able to select the one I want to open. Thanks in advance for help on this if it is possible?

Don

**Comment by ttscoff** [comment]
You should just be able to create a basic shell script that does the cd 
and ls, then run it as a variable setting:

result = $ scripts/myscript.sh

Then ${result} would contain the result of the ls command.

-Brett

**Reply by dmv1** [reply]
Thank you, this did work well, but it raise another question. How can I wait for this result to return before I move onto running a snippet? 

For example, I open Terminal and pass in the script as follows, but the Choice Dialog appears before the Terminal script ran. I have added sequence: sequential to Frontmatter but still seems to run the snippet before the app. I have also tried adding a delay to the Terminal as well as the scripts line (using ~5) but this does not work either. Sorry if this is a basic question but I can not seem to figure out how to 'await' the result of the script running.

Thanks in advance, I am really enjoying working with Bunch!
Thanks for the help
Don

Example Bunch: 

Terminal
files = $ scripts/myscript.sh

<<#Open File
___
[Open File]
?[${files}|raw] "Select File to Open"

**Comment by ttscoff** [comment]
Sequential execution of scripts and dialogs is a little wonky, as Bunch wants to resolve all dialogs before processing (which I should probably fix). You might try using “from script” frontmatter, where it returns the variable in YAML format. That should always execute before dialogs.

Side note, the |raw should be inside the curly brackets.

Thanks!

Brett

**Comment by ttscoff** [comment]
https://bunchapp.co/docs/bunch-files/frontmatter/#dynamicfrontmatter

**Comment by dmv1** [comment]
Thanks so much, I will try adding this in frontmatter
I appreciate the help and the app!! No matter if snippets are wonky, I appreciate what you have built and see this to become a very useful automation tool as I move forward .

Also, thanks about 'raw'!

Don
